// product-a/esds-build-config.js
'use strict';
const cssBeautify = require('js-beautify').css,
    cssBeautifyOptions = {
        newline_between_rules: true,
        selector_separator_newline: true,
        preserve_newlines: true,
        max_preserve_newlines: 1
    },
    htmlBeautify = require('js-beautify').html,
    htmlBeautifyOptions = {
        allowed_file_extensions: ["htm", "html", "xhtml", "shtml", "xml", "svg"],
        brace_style: "collapse",
        end_with_newline: false,
        indent_char: " ",
        indent_handlebars: false,
        indent_inner_html: true,
        indent_size: 4,
        max_preserve_newlines: 0,
        preserve_newlines: false,
        unformatted: ["img", "code", "pre", "sub", "sup", "em", "strong", "b", "i", "u", "strike", "big", "small", "pre"],
        wrap_line_length: 0
    },
    stripIndent = require('strip-indent');

module.exports = {
    codeNamespace: 'product-a',
    manageNunjucksEnv: function(env) {
        env.addFilter('isarray', function(obj) {
          return Array.isArray(obj);
        });

        env.addFilter('isnumber', function(obj) {
          return !isNaN(obj);
        });

        env.addFilter('htmlbeautify', function(string) {
            return htmlBeautify(string, htmlBeautifyOptions);
        });

        env.addFilter('cssbeautify', function(string) {
            return cssBeautify(string, cssBeautifyOptions);
        });

        env.addFilter('stripindent', function(string){
            return stripIndent(string);
        });
    }
};
